VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/EqualsAndGetHashcodeNotImplementedTogether/EqualsAndGetHashcodeNotImplementedTogetherCodeFix.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.EqualsAndGetHashcodeNotImplementedTogether
+{
+    [ExportCodeFixProvider(DiagnosticId.EqualsAndGetHashcodeNotImplementedTogether + "CF", LanguageNames.CSharp), Shared]
+    public class EqualsAndGetHashcodeNotImplementedTogetherCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(EqualsAndGetHashcodeNotImplementedTogetherAnalyzer.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var statement = root.FindNode(diagnosticSpan);
+
+            if (bool.Parse(diagnostic.Properties["IsEqualsImplemented"]))
```

Comment:
You can compress this `if` by moving the entire `context.RegisterCodeFix()` right outside of it and just taking care of the code fix title inside the if (perhaps even by using a ternary if).

Something like this:

	var message = bool.Parse(diagnostic.Properties["IsEqualsImplemented"])
					? "GetHashCode()"
					: "Equals()";

	context.RegisterCodeFix(
		 CodeAction.Create(string.Format(VSDiagnosticsResources.EqualsAndGetHashcodeNotImplementedTogetherCodeFixTitle, message),
		 x => ImplementGetHashCodeAsync(context.Document, root, statement),
		 EqualsAndGetHashcodeNotImplementedTogetherAnalyzer.Rule.Id),
		 diagnostic);